### ISSUE TITLE
Module azure_rm_publicipaddress crashes when ip_tags parameter is set

### ISSUE THREAD

**Issue by razvanpaul-mielcioiu** [ansible/ansible_collections_azure#711]
##### SUMMARY

Module azure_rm_publicipaddress crashes when ip_tags parameter is set by throwing AttributeError: 'dict' object has no attribute 'type'.
This works if I don't set ip_tags.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_publicipaddress

##### ANSIBLE VERSION

```paste below
ansible 2.10.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/var/lib/awx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Aug 24 2020, 17:57:11) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
```

##### COLLECTION VERSION

```paste below
# /var/lib/awx/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.10.0
```

##### CONFIGURATION

```paste below
(empty output from command)
```

##### OS / ENVIRONMENT

localhost connection

##### STEPS TO REPRODUCE

```yaml
 - name: Create a public ip address
   azure_rm_publicipaddress:
     resource_group: "myResourceGroup"
     name: "myPublicipName"
     allocation_method: static
     sku: Standard
     version: ipv4
     ip_tags:
       - ip_tag_type: RoutingPreference
         value: Internet
```

##### EXPECTED RESULTS

public-ip created successfully

##### ACTUAL RESULTS

module throws an internal exception

```paste below
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\\n  File \\"/root/.ansible/tmp/ansible-tmp-1639755477.4218478-24681-41655645874198/AnsiballZ_azure_rm_publicipaddress.py\\", line 102, in <module>\\n    _ansiballz_main()\\n  File \\"/root/.ansible/tmp/ansible-tmp-1639755477.4218478-24681-41655645874198/AnsiballZ_azure_rm_publicipaddress.py\\", line 94, in _ansiballz_main\\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\\n  File \\"/root/.ansible/tmp/ansible-tmp-1639755477.4218478-24681-41655645874198/AnsiballZ_azure_rm_publicipaddress.py\\", line 40, in invoke_module\\n    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_publicipaddress', init_globals=None, run_name='__main__', alter_sys=True)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 205, in run_module\\n    return _run_module_code(code, init_globals, run_name, mod_spec)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 96, in _run_module_code\\n    mod_name, mod_spec, pkg_name, script_name)\\n  File \\"/usr/lib64/python3.6/runpy.py\\", line 85, in _run_code\\n    exec(code, run_globals)\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipaddress.py\\", line 419, in <module>\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipaddress.py\\", line 415, in main\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipaddress.py\\", line 280, in __init__\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\\", line 469, in __init__\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipaddress.py\\", line 370, in exec_module\\n  File \\"/tmp/ansible_azure_rm_publicipaddress_payload_0f5o_hb9/ansible_azure_rm_publicipaddress_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_publicipaddress.py\\", line 370, in <listcomp>\\nAttributeError: 'dict' object has no attribute 'type'\\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Comment by SSPJ** [comment]
I am experiencing this as well.

azure.azcollection 1.12.0

**Comment by Fred-sun** [comment]
fixes by #908
